src/main/java/org/killbill/billing/client/KillBillHttpClient.java

```diff
@@ -103,9 +117,16 @@ public KillBillHttpClient(final String kbServerUrl, final String username, final
 
 
         final AsyncHttpClientConfig.Builder cfg = new AsyncHttpClientConfig.Builder();
-        cfg.setConnectTimeout(DEFAULT_HTTP_TIMEOUT_SEC * 1000);
-        cfg.setRequestTimeout(DEFAULT_HTTP_TIMEOUT_SEC * 1000);
-        cfg.setReadTimeout(DEFAULT_HTTP_TIMEOUT_SEC * 1000);
+
+        if (connectTimeOut != null && readTimeOut != null && requestTimeout != null) {
```

Comment:
I exposed everything that was set in the constructor as a parameter. I think it's a good idea that the user of the client can decided if they want to set them independently. At the same time I am not sure about the meaning of the request timeout, because in the Http clients I am used to you normally just set the read and connect timeout.